# Casusopdracht - Films

**Klas:** V2B
**Studenten:** 
- Roan Gaasbeek
- Mathijs
- Luc

### Inhoud ###
- [Onderzoeksvragen](1.-Onderzoeksvragen)

In [196]:
import numpy as np
import pandas as pd

# 1. Onderzoeksvragen
De volgende onderzoeksvragen worden behandeld in deze opdracht:

- In hoeverre is de omzet van een film te voorspellen op basis van de populariteit op $Facebook$ en $IMDB$?
- ** zelf nog een vraag bedenken met betrekking op een zelf bijgezochte dataset **
- In hoeverre is het mogelijk om logische clusters te vinden o.b.v. onder andere budget en omzet? Denk aan blockbusters (hoog budget en hoge omzet), flops (hoog budget en lage omzet) of cultfilms (laag budget en hoge omzet). Je mag extra features toevoegen. 

# 2. Data understanding

De eesrte stap in deze opdracht is de **data understanding** fase van **CRISP-DM** dit houdt in:
- Data Collection
- Data Exploration & Analysis

#### 2.1 Data Collection
- de meegekregen movie.csv
- de Bechdeltest

In [197]:
# Dit is de casus dataset
dataset = pd.read_csv('movie.csv')
# dit is de gekozen externe dataset
# komt nog

#### 2.2 Data Exploration & Analysis

In [198]:
# Om een eerste blik van de dataset te krijgen gebruiken we de head functie
dataset.head()

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4    NaN        Doug Walker                     NaN       NaN   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    131.0                     NaN        Rob Walker   

   actor_1_facebook_likes        gross                           genres  ...  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi  ...   
1                 40000.0  309404152.0         Action|Adventure|Fantasy  ...   
2                 11000.0  200074175.0        Action|Adventure|Thriller  ...   
3                 27000.0  448130642.0                  Action|Thriller  ...   
4                   131.0          NaN                      Documentary  ...   

  num_user_for_reviews language  country  content_rating       budget  \
0               3054.0  English      USA           PG-13  237000000.0   
1               1238.0  English      USA           PG-13  300000000.0   
2                994.0  English       UK           PG-13  245000000.0   
3               2701.0  English      USA           PG-13  250000000.0   
4                  NaN      NaN      NaN             NaN          NaN   

   title_year actor_2_facebook_likes imdb_score  aspect_ratio  \
0      2009.0                  936.0        7.9          1.78   
1      2007.0                 5000.0        7.1          2.35   
2      2015.0                  393.0        6.8          2.35   
3      2012.0                23000.0        8.5          2.35   
4         NaN                   12.0        7.1           NaN   

  movie_facebook_likes  
0                33000  
1                    0  
2                85000  
3               164000  
4                    0  

[5 rows x 28 columns]

In [199]:
# dit is de lijst met de features van onze dataset
dataset.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

### features en targets
hieronder een lijst van de features met meetniveaus die naar onze mening interessant zijn voor dit onderzoek
- director_name (Nominaal)
- actor_1_name  (Nominaal)
- actor_2_name  (Nominaal)
- actor_3_name  (Nominaal)
- cast_facebook_likes   (Discreet)
- budget    (Continu)
- imdb_score    (Continu)
- movie_facebook_likes  (Discreet)

en de targetvariabel is al bepaald en dat is de opbrengst van de film
- gross (Continu)

In [200]:
# we gebruiken deze functie om een samenvatting van de features te krijgen.
dataset.describe()

num_critic_for_reviews     duration  director_facebook_likes  \
count             4993.000000  5028.000000              4939.000000   
mean               140.194272   107.201074               686.509212   
std                121.601675    25.197441              2813.328607   
min                  1.000000     7.000000                 0.000000   
25%                 50.000000    93.000000                 7.000000   
50%                110.000000   103.000000                49.000000   
75%                195.000000   118.000000               194.500000   
max                813.000000   511.000000             23000.000000   

       actor_3_facebook_likes  actor_1_facebook_likes         gross  \
count             5020.000000             5036.000000  4.159000e+03   
mean               645.009761             6560.047061  4.846841e+07   
std               1665.041728            15020.759120  6.845299e+07   
min                  0.000000                0.000000  1.620000e+02   
25%                133.000000              614.000000  5.340988e+06   
50%                371.500000              988.000000  2.551750e+07   
75%                636.000000            11000.000000  6.230944e+07   
max              23000.000000           640000.000000  7.605058e+08   

       num_voted_users  cast_total_facebook_likes  facenumber_in_poster  \
count     5.043000e+03                5043.000000           5030.000000   
mean      8.366816e+04                9699.063851              1.371173   
std       1.384853e+05               18163.799124              2.013576   
min       5.000000e+00                   0.000000              0.000000   
25%       8.593500e+03                1411.000000              0.000000   
50%       3.435900e+04                3090.000000              1.000000   
75%       9.630900e+04               13756.500000              2.000000   
max       1.689764e+06              656730.000000             43.000000   

       num_user_for_reviews        budget   title_year  \
count           5022.000000  4.551000e+03  4935.000000   
mean             272.770808  3.975262e+07  2002.470517   
std              377.982886  2.061149e+08    12.474599   
min                1.000000  2.180000e+02  1916.000000   
25%               65.000000  6.000000e+06  1999.000000   
50%              156.000000  2.000000e+07  2005.000000   
75%              326.000000  4.500000e+07  2011.000000   
max             5060.000000  1.221550e+10  2016.000000   

       actor_2_facebook_likes   imdb_score  aspect_ratio  movie_facebook_likes  
count             5030.000000  5043.000000   4714.000000           5043.000000  
mean              1651.754473     6.442138      2.220403           7525.964505  
std               4042.438863     1.125116      1.385113          19320.445110  
min                  0.000000     1.600000      1.180000              0.000000  
25%                281.000000     5.800000      1.850000              0.000000  
50%                595.000000     6.600000      2.350000            166.000000  
75%                918.000000     7.200000      2.350000           3000.000000  
max             137000.000000     9.500000     16.000000         349000.000000

In [201]:
# met de functie isnull() kunnen we kijken hoeveel van de data we missen 
totaalnull = dataset.isnull().sum()
totaalnull

color                         19
director_name                104
num_critic_for_reviews        50
duration                      15
director_facebook_likes      104
actor_3_facebook_likes        23
actor_2_name                  13
actor_1_facebook_likes         7
gross                        884
genres                         0
actor_1_name                   7
movie_title                    0
num_voted_users                0
cast_total_facebook_likes      0
actor_3_name                  23
facenumber_in_poster          13
plot_keywords                153
movie_imdb_link                0
num_user_for_reviews          21
language                      14
country                        5
content_rating               303
budget                       492
title_year                   108
actor_2_facebook_likes        13
imdb_score                     0
aspect_ratio                 329
movie_facebook_likes           0
dtype: int64

### Analyse van feature- en targetvariabelen

Te onderzoeken featurevariabelen:  
- movie_facebook_likes
- imdb_score
- budget

Te onderzoeken targetvariabele:  
- gross

In [202]:
# Kolommen selecteren
kolommen = ['movie_facebook_likes', 'imdb_score', 'budget', 'gross']
data_analyse = dataset[kolommen]
data_analyse


movie_facebook_likes  imdb_score       budget        gross
0                    33000         7.9  237000000.0  760505847.0
1                        0         7.1  300000000.0  309404152.0
2                    85000         6.8  245000000.0  200074175.0
3                   164000         8.5  250000000.0  448130642.0
4                        0         7.1          NaN          NaN
...                    ...         ...          ...          ...
5038                    84         7.7          NaN          NaN
5039                 32000         7.5          NaN          NaN
5040                    16         6.3       1400.0          NaN
5041                   660         6.3          NaN      10443.0
5042                   456         6.6       1100.0      85222.0

[5043 rows x 4 columns]

In [203]:
# Centrum- en spreidingsmaten
data_analyse.describe()

movie_facebook_likes   imdb_score        budget         gross
count           5043.000000  5043.000000  4.551000e+03  4.159000e+03
mean            7525.964505     6.442138  3.975262e+07  4.846841e+07
std            19320.445110     1.125116  2.061149e+08  6.845299e+07
min                0.000000     1.600000  2.180000e+02  1.620000e+02
25%                0.000000     5.800000  6.000000e+06  5.340988e+06
50%              166.000000     6.600000  2.000000e+07  2.551750e+07
75%             3000.000000     7.200000  4.500000e+07  6.230944e+07
max           349000.000000     9.500000  1.221550e+10  7.605058e+08

### Waarnemingen centrum- en spreidingsmaten

- De kolommen movie_facebook_likes en imdb_score hebben beide 5043 waarden, maar budget en gross hebben minder waarden (4551 en 4159 waarden).  
We zullen hier een oplossing voor moeten vinden, door bijvoorbeeld de rijen met missende waarden te droppen van de dataset.
- Q1 voor movie_facebook_likes is 0, dit betekent dat ten minste 25% van de films 0 facebook likes hebben. Dit komt waarschijnlijk omdat deze films geen facebook-pagina hebben.


In [204]:
# Uitschieters bepalen
for kolom in data_analyse:

    # Q1, Q3 en IQR bepalen
    q1 = data_analyse[kolom].quantile(0.25)
    q3 = data_analyse[kolom].quantile(0.75)
    iqr = q3 - q1

    # Limieten uitschieters berekenen
    klein_laag = q1 - 1.5 * iqr
    klein_hoog = q3 + 1.5 * iqr
    extreem_laag = q1 - 3 * iqr
    extreem_hoog = q3 + 3 * iqr

    print(data_analyse[kolom].name)

    # Kleine uitschieters (1.5 - 3 IQR)
    kleine_uitschieters = data_analyse[kolom][((data_analyse[kolom] < klein_laag) & (data_analyse[kolom] > extreem_laag)) | ((data_analyse[kolom] > klein_hoog) & (data_analyse[kolom] < extreem_hoog))]
    print(f"Kleine uitschieters: {kleine_uitschieters.count()}")
 
    # Extreme uitschieters (3+ IQR)
    extreme_uitschieters = data_analyse[kolom][(data_analyse[kolom] < extreem_laag) | (data_analyse[kolom] > extreem_hoog)]
    print(f"Extreme uitschieters: {extreme_uitschieters.count()}")

movie_facebook_likes
Kleine uitschieters: 181
Extreme uitschieters: 930
imdb_score
Kleine uitschieters: 120
Extreme uitschieters: 0
budget
Kleine uitschieters: 200
Extreme uitschieters: 118
gross
Kleine uitschieters: 187
Extreme uitschieters: 119


# 3. Data preparation
De volgende stap in deze is de data preparation fase van $CRISP-DM$ dit houdt in:
- Data Cleaning
- Data Processing

#### 3.1 Data Processing
*uitleg over wat in dit stuk behandeld wordt*


In [205]:
dataset.drop(columns=['color', 'actor_1_facebook_likes', 'actor_2_facebook_likes', 'actor_3_facebook_likes', 'duration', 'genres', 'num_voted_users', 'facenumber_in_poster','plot_keywords', 'num_user_for_reviews','num_critic_for_reviews', 'language', 'country', 'content_rating', 'title_year','aspect_ratio'], inplace=True)

# we hebben de link voor de imdb pagina niet nodig maar hierin staat wel de imdb_id die we later kunnen gebruiken om een externe dataset te koppelen.
dataset['movie_imdb_link'] = dataset['movie_imdb_link'].str.slice(start=28, stop=-17)

dataset.rename(columns={'director_name':'director', 'movie_title':'title', 'movie_imdb_link':'imdb_id'}, inplace=True)

dataset.head()

director  director_facebook_likes      actor_2_name        gross  \
0      James Cameron                      0.0  Joel David Moore  760505847.0   
1     Gore Verbinski                    563.0     Orlando Bloom  309404152.0   
2         Sam Mendes                      0.0      Rory Kinnear  200074175.0   
3  Christopher Nolan                  22000.0    Christian Bale  448130642.0   
4        Doug Walker                    131.0        Rob Walker          NaN   

      actor_1_name                                              title  \
0      CCH Pounder                                            Avatar    
1      Johnny Depp          Pirates of the Caribbean: At World's End    
2  Christoph Waltz                                           Spectre    
3        Tom Hardy                             The Dark Knight Rises    
4      Doug Walker  Star Wars: Episode VII - The Force Awakens    ...   

   cast_total_facebook_likes          actor_3_name  imdb_id       budget  \
0                       4834             Wes Studi  0499549  237000000.0   
1                      48350        Jack Davenport  0449088  300000000.0   
2                      11700      Stephanie Sigman  2379713  245000000.0   
3                     106759  Joseph Gordon-Levitt  1345836  250000000.0   
4                        143                   NaN  5289954          NaN   

   imdb_score  movie_facebook_likes  
0         7.9                 33000  
1         7.1                     0  
2         6.8                 85000  
3         8.5                164000  
4         7.1                     0

*zullen we hier ook nog even de volgorde van kolommen aanpassen om het wat leesbaarder te maken*

#### 3.2 Data Cleaning



In [206]:
dataset.describe(include="all")
totaal_films = dataset['color'].count()

kleur_films = dataset[dataset['color'] == 'Color'].shape[0]

zw_films = dataset[dataset['color'] != 'Color'].shape[0]  

percentage_kleur = (kleur_films / totaal_films) * 100
percentage_zw = (zw_films / totaal_films) * 100

print(f"Percentage gekleurde films: {percentage_kleur:.2f}%")
print(f"Percentage zwart-wit films: {percentage_zw:.2f}%")


KeyError: 'color'

# Color Feature

In [14]:
#Zoals hierboven zijn bijna 96% van de films in kleur, daarom ervoor gekozen om deze kolom te droppen.
dataset.drop(['color'], axis = 1)


KeyError: "['color'] not found in axis"

#  Aspect Ratio Feature

In [15]:
#Aspect ratio is niet relevant
dataset.drop(['aspect_ratio'], axis = 1)


KeyError: "['aspect_ratio'] not found in axis"

#  Movie Imdb Link Feature

In [36]:
#Movie Imdb Link is ook niet relevant
dataset.drop(['movie_imdb_link'], axis = 1)


color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4       NaN        Doug Walker                     NaN       NaN   
...     ...                ...                     ...       ...   
5038  Color        Scott Smith                     1.0      87.0   
5039  Color                NaN                    43.0      43.0   
5040  Color   Benjamin Roberds                    13.0      76.0   
5041  Color        Daniel Hsia                    14.0     100.0   
5042  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                         0.0                   855.0  Joel David Moore   
1                       563.0                  1000.0     Orlando Bloom   
2                         0.0                   161.0      Rory Kinnear   
3                     22000.0                 23000.0    Christian Bale   
4                       131.0                     NaN        Rob Walker   
...                       ...                     ...               ...   
5038                      2.0                   318.0     Daphne Zuniga   
5039                      NaN                   319.0     Valorie Curry   
5040                      0.0                     0.0     Maxwell Moody   
5041                      0.0                   489.0     Daniel Henney   
5042                     16.0                    16.0  Brian Herzlinger   

      actor_1_facebook_likes        gross                           genres  \
0                     1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi   
1                    40000.0  309404152.0         Action|Adventure|Fantasy   
2                    11000.0  200074175.0        Action|Adventure|Thriller   
3                    27000.0  448130642.0                  Action|Thriller   
4                      131.0          NaN                      Documentary   
...                      ...          ...                              ...   
5038                   637.0          NaN                     Comedy|Drama   
5039                   841.0          NaN     Crime|Drama|Mystery|Thriller   
5040                     0.0          NaN            Drama|Horror|Thriller   
5041                   946.0      10443.0             Comedy|Drama|Romance   
5042                    86.0      85222.0                      Documentary   

      ... num_user_for_reviews language  country  content_rating       budget  \
0     ...               3054.0  English      USA           PG-13  237000000.0   
1     ...               1238.0  English      USA           PG-13  300000000.0   
2     ...                994.0  English       UK           PG-13  245000000.0   
3     ...               2701.0  English      USA           PG-13  250000000.0   
4     ...                  NaN      NaN      NaN             NaN          NaN   
...   ...                  ...      ...      ...             ...          ...   
5038  ...                  6.0  English   Canada             NaN          NaN   
5039  ...                359.0  English      USA           TV-14          NaN   
5040  ...                  3.0  English      USA             NaN       1400.0   
5041  ...                  9.0  English      USA           PG-13          NaN   
5042  ...                 84.0  English      USA              PG       1100.0   

      title_year actor_2_facebook_likes  imdb_score aspect_ratio  \
0         2009.0                  936.0         7.9         1.78   
1         2007.0                 5000.0         7.1         2.35   
2         2015.0                  393.0         6.8         2.35   
3         2012.0                23000.0         8.5         2.35   
4            NaN                   12.

#  Gross Target

In [46]:
#Hier wordt het gemiddelde van de gross waardes die er zijn berekend
gemiddelde_gross = dataset['gross'].mean()
print(f"Gemiddelde gross: ${gemiddelde_gross:.2f}")


Gemiddelde gross: $48468407.53


In [62]:

dataset.fillna({"gross" :48468407.53 })

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4       NaN        Doug Walker                     NaN       NaN   
...     ...                ...                     ...       ...   
5038  Color        Scott Smith                     1.0      87.0   
5039  Color                NaN                    43.0      43.0   
5040  Color   Benjamin Roberds                    13.0      76.0   
5041  Color        Daniel Hsia                    14.0     100.0   
5042  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                         0.0                   855.0  Joel David Moore   
1                       563.0                  1000.0     Orlando Bloom   
2                         0.0                   161.0      Rory Kinnear   
3                     22000.0                 23000.0    Christian Bale   
4                       131.0                     NaN        Rob Walker   
...                       ...                     ...               ...   
5038                      2.0                   318.0     Daphne Zuniga   
5039                      NaN                   319.0     Valorie Curry   
5040                      0.0                     0.0     Maxwell Moody   
5041                      0.0                   489.0     Daniel Henney   
5042                     16.0                    16.0  Brian Herzlinger   

      actor_1_facebook_likes         gross                           genres  \
0                     1000.0  7.605058e+08  Action|Adventure|Fantasy|Sci-Fi   
1                    40000.0  3.094042e+08         Action|Adventure|Fantasy   
2                    11000.0  2.000742e+08        Action|Adventure|Thriller   
3                    27000.0  4.481306e+08                  Action|Thriller   
4                      131.0  4.846841e+07                      Documentary   
...                      ...           ...                              ...   
5038                   637.0  4.846841e+07                     Comedy|Drama   
5039                   841.0  4.846841e+07     Crime|Drama|Mystery|Thriller   
5040                     0.0  4.846841e+07            Drama|Horror|Thriller   
5041                   946.0  1.044300e+04             Comedy|Drama|Romance   
5042                    86.0  8.522200e+04                      Documentary   

      ... num_user_for_reviews language  country  content_rating       budget  \
0     ...               3054.0  English      USA           PG-13  237000000.0   
1     ...               1238.0  English      USA           PG-13  300000000.0   
2     ...                994.0  English       UK           PG-13  245000000.0   
3     ...               2701.0  English      USA           PG-13  250000000.0   
4     ...                  NaN      NaN      NaN             NaN     500000.0   
...   ...                  ...      ...      ...             ...          ...   
5038  ...                  6.0  English   Canada             NaN    7000000.0   
5039  ...                359.0  English      USA           TV-14   29000000.0   
5040  ...                  3.0  English      USA             NaN       1400.0   
5041  ...                  9.0  English      USA           PG-13   15000000.0   
5042  ...                 84.0  English      USA              PG       1100.0   

      title_year actor_2_facebook_likes imdb_score  aspect_ratio  \
0         2009.0                  936.0        7.9          1.78   
1         2007.0                 5000.0        7.1          2.35   
2         2015.0                  393.0        6.8          2.35   
3         2012.0                23000.0        8.5          2.35   
4            NaN          

#  Budget Target

In [50]:
gemiddelde_budget = dataset['budget'].mean()
print(f"Gemiddelde budget: ${gemiddelde_budget:.2f}")


Gemiddelde budget: $37916434.79


In [57]:
dataset.fillna({"budget" :48468407.53 })

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4       NaN        Doug Walker                     NaN       NaN   
...     ...                ...                     ...       ...   
5038  Color        Scott Smith                     1.0      87.0   
5039  Color                NaN                    43.0      43.0   
5040  Color   Benjamin Roberds                    13.0      76.0   
5041  Color        Daniel Hsia                    14.0     100.0   
5042  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                         0.0                   855.0  Joel David Moore   
1                       563.0                  1000.0     Orlando Bloom   
2                         0.0                   161.0      Rory Kinnear   
3                     22000.0                 23000.0    Christian Bale   
4                       131.0                     NaN        Rob Walker   
...                       ...                     ...               ...   
5038                      2.0                   318.0     Daphne Zuniga   
5039                      NaN                   319.0     Valorie Curry   
5040                      0.0                     0.0     Maxwell Moody   
5041                      0.0                   489.0     Daniel Henney   
5042                     16.0                    16.0  Brian Herzlinger   

      actor_1_facebook_likes        gross                           genres  \
0                     1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi   
1                    40000.0  309404152.0         Action|Adventure|Fantasy   
2                    11000.0  200074175.0        Action|Adventure|Thriller   
3                    27000.0  448130642.0                  Action|Thriller   
4                      131.0          NaN                      Documentary   
...                      ...          ...                              ...   
5038                   637.0          NaN                     Comedy|Drama   
5039                   841.0          NaN     Crime|Drama|Mystery|Thriller   
5040                     0.0          NaN            Drama|Horror|Thriller   
5041                   946.0      10443.0             Comedy|Drama|Romance   
5042                    86.0      85222.0                      Documentary   

      ... num_user_for_reviews language  country  content_rating       budget  \
0     ...               3054.0  English      USA           PG-13  237000000.0   
1     ...               1238.0  English      USA           PG-13  300000000.0   
2     ...                994.0  English       UK           PG-13  245000000.0   
3     ...               2701.0  English      USA           PG-13  250000000.0   
4     ...                  NaN      NaN      NaN             NaN     500000.0   
...   ...                  ...      ...      ...             ...          ...   
5038  ...                  6.0  English   Canada             NaN    7000000.0   
5039  ...                359.0  English      USA           TV-14   29000000.0   
5040  ...                  3.0  English      USA             NaN       1400.0   
5041  ...                  9.0  English      USA           PG-13   15000000.0   
5042  ...                 84.0  English      USA              PG       1100.0   

      title_year actor_2_facebook_likes imdb_score  aspect_ratio  \
0         2009.0                  936.0        7.9          1.78   
1         2007.0                 5000.0        7.1          2.35   
2         2015.0                  393.0        6.8          2.35   
3         2012.0                23000.0        8.5          2.35   
4            NaN                   12.

# Datatypes